File: js-archive/add-strings.js
var addStrings = function (num1, num2) {
  const n1 = num1.length
  const n2 = num2.length
  if (n1 > n2) {
    for (let i = 0; i < n1 - n2; i++) {
      num2 = '0' + num2
    }
  } else {
    for (let i = 0; i < n2 - n1; i++) {
      num1 = '0' + num1
    }
  }
  let mem = 0
  let ans = ''
  for (let i = num1.length - 1; i >= 0; i--) {
    let sum = Number(num1[i]) + Number(num2[i]) + mem
    if (sum <= 9) {
      ans += sum
      mem = 0
    } else {
      ans += (sum % 10)
      mem = 1
    }
  }
  if (mem > 0) {
    ans += mem
  }

  return (ans.split('').reverse().join(''))
};

console.log(addStrings("11", "123"))

File: add-strings.go
func addStrings(num1 string, num2 string) string {
    for len(num1) < len(num2) {num1 = "0" + num1}
    for len(num2) < len(num1) {num2 = "0" + num2}

    num1 = rev(num1); num2 = rev(num2); ans:=""; rem:=0

    for i:=0; i < len(num1); i++ {
        n1, _ := strconv.Atoi(string(num1[i]))
        n2, _ := strconv.Atoi(string(num2[i]))
        sum:=rem + n1 + n2
        if (sum > 9) {
            rem = 1
            sum%=10
        } else {
            rem = 0
        }

        ans+=strconv.Itoa(sum)
    }

    if (rem == 1) {ans+="1"}

    return rev(ans)
}

func rev(s string) string {
    ans:=""
    for i:=len(s) - 1; i>=0; i-- {
        ans+=string(s[i])
    }

    return ans
}